src/dolphin/unwrap/_unwrap.py

```diff
@@ -272,12 +286,47 @@ def unwrap(
             output_filename=combined_mask_file,
         )
 
+    if run_goldstein:
+        suf = Path(unw_filename).suffix
+        if suf == ".tif":
+            driver = "GTiff"
+            opts = list(io.DEFAULT_TIFF_OPTIONS)
+        else:
+            driver = "ENVI"
+            opts = list(io.DEFAULT_ENVI_OPTIONS)
+
+        # If we're running Goldstein filtering, the intermediate
+        # filtered/unwrapped rasters are temporary rasters in the scratch dir.
+        filt_ifg_filename = (
+            Path(scratchdir or ".") / Path(ifg_filename).with_suffix(".filt" + suf).name
+        )
+        scratch_unw_filename = Path(unw_filename).with_suffix(".filt.unw" + suf)
+
+        ifg = io.load_gdal(ifg_filename)
+        logger.info(f"Goldstein filtering {ifg_filename} -> {filt_ifg_filename}")
+        filt_ifg = goldstein(ifg, alpha=alpha)
+        logger.info(f"Writing filtered output to {filt_ifg_filename}")
+        io.write_arr(
+            arr=filt_ifg,
+            output_name=filt_ifg_filename,
+            like_filename=ifg_filename,
+            driver=driver,
+            options=opts,
+        )
+        unwrapper_ifg_filename = filt_ifg_filename
+        unwrapper_unw_filename = scratch_unw_filename
+    else:
+        unwrapper_ifg_filename = Path(ifg_filename)
+        unwrapper_unw_filename = Path(unw_filename)
```

Comment:
Note for @mirzaees and @taliboliver - this is where it could make sense to add the interpolation logic.
If this section gets unwieldy, perhaps we could break it into a helper function that returns the unwrapper `ifg_filename` and `unw_filename`